internal/pkg/manifest/validate.go

```diff
@@ -1856,6 +1860,62 @@ func populateALBPortsAndValidate(containerNameFor map[uint16]string, opts valida
 	return nil
 }
 
+func populateNLBPortsAndValidate(containerNameFor map[uint16]string, opts validateExposedPortsOpts) error {
+	if opts.nlb == nil {
+		return nil
+	}
+	nlb := opts.nlb
+	if nlb.Port == nil {
+		return nil
+	}
+	nlbPort, protocol, err := ParsePortMapping(nlb.Port)
+	if err != nil {
+		return err
+	}
+	if protocol != nil {
+		protocolVal := aws.StringValue(protocol)
+		var isValidProtocol bool
+		for _, valid := range nlbValidProtocols {
+			if strings.EqualFold(protocolVal, valid) {
+				isValidProtocol = true
+				break
+			}
+		}
+		if !isValidProtocol {
+			return fmt.Errorf(`invalid protocol %s; valid protocols include %s`, protocolVal, english.WordSeries(nlbValidProtocols, "and"))
```

Comment:
Isn't NLB port already validated [here](https://github.com/aws/copilot-cli/blob/856e9dd0cdb46077e3db77e4745d0e60c106dd8a/internal/pkg/manifest/validate.go#L806)?

Reply:
good catch, I will remove this part as it is not required.